File: frontend/bff/src/application/mandates/queries/list-mandates.query.ts
import type { MandateStatus } from '../../../cross-cutting/enums/mandate-status.js';
import type { PageRequest } from '../../../cross-cutting/paging/page.js';
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface ListMandatesQuery {
  context: RequestContext;
  policyId: string | null;
  status: MandateStatus | null;
  page: PageRequest;
}

File: backend/core-service/src/Fix.Application/UseCases/Mandates/Queries/ListMandatesQuery.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Mandates.Dtos;
using Fix.Domain.Abstractions;
using Fix.Domain.AggregateRoots.Mandates;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Mandates.Queries;

[RequireRole(RoleCodes.ViewMandate)]
public sealed record ListMandatesQuery(
    Guid UserId,
    Guid OrganizationId,
    Guid? PolicyId,
    MandateStatus? Status,
    int Page,
    int PageSize)
    : IQuery<PagedList<MandateDto>>, IOrganizationRequest;

